src/http.c

```diff
@@ -78,23 +78,23 @@ int http_connect( http_t *conn, int proto, char *proxy, char *host, int port, ch
 		return( 0 );
 
 	//For SSl through proxy
-	if( conn->proxy && PROTO_IS_SECURE(conn->proto))
+	if( conn->proxy && PROTO_IS_SECURE(conn->proto) )
 	  {
-	    char proxybuffer[2000];
-	    sprintf(proxybuffer,"CONNECT %s HTTP/1.1\r\nHost: %s\r\nUser-Agent: Axel\r\n\r\n",conn->host,conn->host);
-
-	    if(write(conn->tcp.fd,proxybuffer,strlen(proxybuffer))==-1)
-	      {
-		return( 0 );
-	      }
-	    if(read(conn->tcp.fd,proxybuffer,2000)==-1)
-	      {
-		return( 0 );
-	      }
-	    if((conn->tcp.ssl = ssl_connect(conn->tcp.fd,conn->headers))==NULL)
-	      return( 0 );
-	  }
-
+		char proxybuffer[USER_AGENT_LEN];
+		int buffer_length = sprintf( proxybuffer, "CONNECT %s HTTP/1.1\r\nHost: %s\r\nUser-Agent: %s\r\n\r\n", conn->host, conn->host, DEFAULT_USER_AGENT );
+	    
+		if( write( conn->tcp.fd, proxybuffer, buffer_length ) == -1 )
```

Comment:
Instead of calling `sprintf` and then `write` you could simply use `dprintf` like this:

	dprintf( conn->tcp.fd, "CONNECT %s HTTP/1.1\r\nHost: %s\r\nUser-Agent: %s\r\n\r\n", conn->host, conn->host, DEFAULT_USER_AGENT );